packages/lib-sourcify/src/lib/logger.ts

```diff
@@ -10,20 +10,43 @@ export const DefaultLogger: ILogger = {
   setLevel(level: number) {
     this.logLevel = level;
   },
-  log(level, msg) {
+  log(level, msg, metadata) {
     if (level <= this.logLevel) {
+      let metadataMsg = '';
+      if (metadata && Object.keys(metadata).length > 0) {
+        metadataMsg += Object.entries(metadata)
+          .map(([key, value]) => {
+            if (typeof value === 'object') {
+              try {
+                value = JSON.stringify(value);
+              } catch (e) {
+                value = 'SerializationError: Unable to serialize object';
+              }
+            }
+            return `${key}=${value}`;
+          })
+          .join('\t');
+      }
       switch (level) {
+        case 0:
+          console.error(msg + ' - ' + metadataMsg);
+          break;
         case 1:
-          console.error(msg);
+          console.warn(msg + ' - ' + metadataMsg);
           break;
         case 2:
-          console.warn(msg);
+          console.info(msg + ' - ' + metadataMsg);
+          break;
+        // Use winston's log levels https://github.com/winstonjs/winston?tab=readme-ov-file#logging-levels
+        // We don't use http (3) and verbose (4)
+        case 5:
+          console.debug(msg + ' - ' + metadataMsg);
           break;
-        case 3:
-          console.info(msg);
+        case 6:
+          console.trace(msg + ' - ' + metadataMsg);
```

Comment:
Is it right to use `console.trace` for `logSilly`? Using `console.trace` will output something like this:

```
Welcome to Node.js v16.20.2.
Type ".help" for more information.
> console.trace("ciao, test")
Trace: ciao, test
    at REPL1:1:9
    at Script.runInThisContext (node:vm:129:12)
    at REPLServer.defaultEval (node:repl:572:29)
    at bound (node:domain:433:15)
    at REPLServer.runBound [as eval] (node:domain:444:12)
    at REPLServer.onLine (node:repl:902:10)
    at REPLServer.emit (node:events:525:35)
    at REPLServer.emit (node:domain:489:12)
    at REPLServer.Interface._onLine (node:readline:491:10)
    at REPLServer.Interface._line (node:readline:868:8)
```

Reply:
My understanding wasn't like this. Good to know 👍